File: market-application/src/main/java/com/example/market/application/command/RegisterProductCommand.java
package com.example.market.application.command;

import com.example.market.domain.catalog.ProductCategory;

import java.time.Instant;

public record RegisterProductCommand(
        String brand,
        String modelName,
        String styleCode,
        ProductCategory category,
        Instant releaseDate,
        String imageUrl
) {}

File: market-application/src/main/kotlin/com/example/market/application/command/RegisterProductCommand.kt
package com.example.market.application.command

import com.example.market.domain.catalog.ProductCategory
import java.time.Instant

@JvmRecord
data class RegisterProductCommand(
    val brand: String,
    val modelName: String,
    val styleCode: String?,
    val category: ProductCategory,
    val releaseDate: Instant?,
    val imageUrl: String?,
)
